tensorflow_quantum/python/util.py

```diff
@@ -519,3 +519,95 @@ def exponential(operators, coefficients=None):
                                   ] + w_dagger + u_dagger
             circuit += cirq.Circuit(exp_circuit)
     return circuit
+
+
+# ==============================================================================
+# Unsigned quantum integer functions.
+# ==============================================================================
+
+
+def projector_on_one(qubit):
+    """Returns the projector on 1 for the given qubit.
+
+    Given a qubit k, the projector onto one can be represented by
+        |1><1|_k = 0.5(I_k - Z_k).
+
+    Args:
+        qubit: A `cirq.GridQubit` on which the projector is supported.
+
+    Returns:
+        `cirq.PauliSum` representing the projector.
+    """
+    if not isinstance(qubit, cirq.GridQubit):
+        raise TypeError("A projector must live on a cirq.GridQubit.")
+    return 0.5 * cirq.I(qubit) - 0.5 * cirq.Z(qubit)
+
+
+def registers_from_precisions(precisions):
```

Comment:
This is still opaque and hard for a user to build on. Suppose I call this function then want to apply some gates to the qubits that correspond to the register that goes with precisions[3]. How do I figure out which qubits those are without reading the source code of this function (even when calling `to_momentum_basis` I get stuck)?

Reply:
This is a function for making a ragged tensor of qubits, which is why the`GridQubit.rect` functionality won't work.  Also, results are meant to be consumed only by functions expecting lists of qubits which are unique, not functions for which individual qubits are special.  For me the idea of designating a set of qubits as a register is that none are supposed be addressed individually by user code

Reply:
> For me the idea of designating a set of qubits as a register is that none are supposed be addressed individually by user code.

I agree. suppose I want to do a QFT on a register. How do I hope to know which qubits correspond to which register without knowing this particular row and column pattern for how this code allocated registers. We don’t want library code having to bake in this assumption.